llama_agi/llama_agi/execution_agent/base.py

```diff
@@ -0,0 +1,41 @@
+from abc import abstractmethod
+from typing import Any, Dict, List, Optional, Union, TypedDict, NotRequired
+
+from langchain.agents.tools import Tool
+from langchain.llms import OpenAI, BaseLLM
+from langchain.chat_models.base import BaseChatModel
+from langchain.chat_models import ChatOpenAI
+
+
+class LlamaAgentPrompts(TypedDict):
+    execution_prompt: NotRequired[str]
+    agent_prefix: NotRequired[str]
+    agent_suffix: NotRequired[str]
+
+
+class BaseExecutionAgent:

```

Comment:
not for this PR, but later we may want to make this even more abstract, so that __init__ is responsibility of the subclasses (e.g. if we want to do a LlamaExecutionAgent that can pass in ServiceContext etc.)

Reply:
Yea definitely, a good change to make at some point